framework/include/materials/DerivativeMaterialInterface.h

```diff
@@ -63,7 +64,7 @@ class DerivativeMaterialInterface : public T
    * Fetch a pointer to a material property if it exists, otherwise return null
    */
   template<typename U>
-  MaterialProperty<U> * getMaterialPropertyPointer(const std::string & name);
+  MaterialProperty<U> * getMaterialPropertyPointer(const std::string & name) __attribute__ ((deprecated));
```

Comment:
I'd rather not add compiler-specific stuff like `__attribute__`... There are MOOSE-specific ways of deprecating stuff that should be used instead.